podpac/core/data/types.py

```diff
@@ -694,7 +697,163 @@ def _find_h5py_keys(obj, keys=[]):
+                    raise ValueError("Zarr time key '%s' not found" % self.timekey)
+            elif dim == 'alt':
+                if self.altkey is None:
+                    raise TypeError("Zarr node 'altkey' is required for dims %s" % self.dims)
+                if self.altkey not in self.group:
+                    raise ValueError("Zarr alt key '%s' not found" % self.altkey)
+
+        # check data key
+        if self.datakey not in self.group:
+            raise ValueError("Zarr data key '%s' not found" % self.datakey)
+
+    @tl.default('group')
+    def _open_group(self):
+        if self.source is None:
+            raise TypeError("Zarr node requires 'source' or 'group'")
+
+        if self.source.startswith('s3://'):
+            root = self.source.strip('s3://')
+            kwargs = {'region_name': self.region_name}
+            s3 = s3fs.S3FileSystem(key=self.access_key_id, secret=self.secret_access_key, client_kwargs=kwargs)
+            s3map = s3fs.S3Map(root=root, s3=s3, check=False)
+            store = s3map
+        else:
+            store = self.source
+        
+        return zarr.open(store, mode='r')
+
+    @common_doc(COMMON_DATA_DOC)
+    def get_native_coordinates(self):
+        """{get_native_coordinates}
+        """
+
+        cs = []
+        for dim in self.dims:
+            if dim == 'lat':
+                cs.append(self.get_lat())
+            elif dim == 'lon':
+                cs.append(self.get_lon())
+            elif dim == 'time':
+                cs.append(self.get_time())
+            elif dim == 'alt':
+                cs.append(self.get_alt())
+
+        return Coordinates(cs, dims=self.dims, crs=self.crs)
+
+    @common_doc(COMMON_DATA_DOC)
+    def get_data(self, coordinates, coordinates_index):
+        """{get_data}
+        """
+        data = self.create_output_array(coordinates)
+        a = self.group[self.datakey][coordinates_index]
+        data.data.ravel()[:] = a.ravel()
+        return data
+
+    def get_lat(self):
+        """
+        Get the native lat coordinates. Subclasses should customize as needed.
+
+        Returns
+        -------
+        lat : array-like or Coordinates1d
+            native latitude coordinates.
+        """
+
+        return self.group[self.latkey]
+
+    def get_lon(self):
+        """
+        Get the native lon coordinates. Subclasses should customize as needed.
+
+        Returns
+        -------
+        lon : array-like or Coordinates1d
+            native latitude coordinates.
+        """
+
+        return self.group[self.lonkey]
+
+    def get_time(self):
+        """
+        Get the native time coordinates. Subclasses should customize as needed.
+
+        Returns
+        -------
+        time : array-like or Coordinates1d
+            native latitude coordinates.
+        """
+
+        values = self.group[self.timekey]
+        if self.cf_time:
+            # values = cftime.num2date(values, self.cf_units, self.cf_calendar)
```

Comment:
This works just as well, but requires the `cftime` library. Would you prefer to add that as a requirement for Zarr nodes, or use it if available and fallback on xarray if necessary, or just leverage xarray like this and be done with it?